client/my-sites/plans-features-main/plan-notices.tsx

```diff
@@ -0,0 +1,136 @@
+import { PLAN_ENTERPRISE_GRID_WPCOM } from '@automattic/calypso-products';
+import { formatCurrency } from '@automattic/format-currency';
+import { useTranslate } from 'i18n-calypso';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import MarketingMessage from 'calypso/components/marketing-message';
+import Notice from 'calypso/components/notice';
+import { getDiscountByName } from 'calypso/lib/discounts';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import {
+	getSitePlanSlug,
+	isCurrentUserCurrentPlanOwner,
+} from 'calypso/state/sites/plans/selectors';
+import { calculatePlanUpgradeCredits } from 'calypso/state/sites/plans/selectors/calculate-plan-upgrade-credits';
+import { isCurrentPlanPaid, isJetpackSite } from 'calypso/state/sites/selectors';
+
+function usePlanUpgradeCreditsDisplay(
```

Comment:
Can we move hooks to a corresponding `/hooks` folder? I can see your intention here, but I prefer to see hooks living separately, exported, and tested fully (so let's also add a test for this as well).